projects/plugins/jetpack/tests/php/modules/masterbar/test-class-featured-post-thumbnail.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Test_Featured_Post_Thumbnail file.
+ * Test Featured_Post_Thumbnail.
+ *
+ *  @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+require_jetpack_file( 'modules/masterbar/featured-post-thumbnail/class-featured-post-thumbnail.php' );
+require_jetpack_file( 'tests/php/modules/masterbar/data/featured-image-post-thumbnail.php' );
+
+/**
+ * Class Test_Featured_Post_Thumbnail
+ */
+class Test_Featured_Post_Thumbnail extends \WP_UnitTestCase {
+
+	/**
+	 * The attachment holds file object.
+	 *
+	 * @var null
+	 */
+	public static $attachment = null;
+
+	/**
+	 * The get attachment endpoint response.
+	 *
+	 * @var \WP_Error|\WP_HTTP_Response|\WP_REST_Response|null
+	 */
+	public static $response = null;
+
+	/**
+	 * Create shared attachment.
+	 *
+	 * @param \WP_UnitTest_Factory $factory Fixture factory.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		$attachment_id = $factory->attachment->create_upload_object( __DIR__ . '/data/files/dummy-file.jpg' );
+
+		self::$attachment = get_post( $attachment_id );
+	}
```

Comment:
Great implementation 👏 ! Do you think we also need a `tearDownAfterClass` function that deletes the post created?

Reply:
I agree. It make sense to remove the created post.

Reply:
Added the `tearDownAfterClass` method and removed created attachment.